### PR TITLE
Bug 1738432: genericoperatorclient: only preserve fields outside of Operator{Spec,Status}

### PR DESCRIPTION
**Pull request by sttts**
Before this, we just used mergo to merge old spec/status with the new value. It had no concept of specified `OperatorSpec/Status` fields and also merged observed configs with mergo. The consequence was that fields were never removed, although the new observed config did not have those fields anymore.

### CODE REVIEW

File: pkg/operator/genericoperatorclient/dynamic_operator_client.go [hunk 1]
```diff
@@ -17,6 +16,8 @@ import (
 	"k8s.io/client-go/informers"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/cache"
+
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
```
Comment (damemi) [inline pkg/operator/genericoperatorclient/dynamic_operator_client.go]: some weird import grouping here

File: pkg/operator/genericoperatorclient/dynamic_operator_client_test.go [hunk 2]
```diff
@@ -29,10 +30,52 @@ func TestSetOperatorSpecFromUnstructured(t *testing.T) {
...
+					"operatorLogLevel":           "",
+					"unsupportedConfigOverrides": nil,
+					"observedConfig":             nil,
+				},
+			},
+		},
+		{
+			name: "replace-rawextensions",
+			in: map[string]interface{}{
+				"spec": map[string]interface{}{
```
Comment (enj) [inline pkg/operator/genericoperatorclient/dynamic_operator_client_test.go]: Do we need some tests with both spec and status set?  Along with metadata?

File: pkg/operator/genericoperatorclient/dynamic_operator_client.go (hunk 1 above)
Comment (sttts) [reply pkg/operator/genericoperatorclient/dynamic_operator_client.go]: fixed

File: pkg/operator/genericoperatorclient/dynamic_operator_client_test.go (hunk 2 above)
Comment (sttts) [reply pkg/operator/genericoperatorclient/dynamic_operator_client_test.go]: Kind of out-of-scope, but added anyway.

**Pull request merged**